pyomo/contrib/mindtpy/single_tree.py

```diff
@@ -88,7 +85,7 @@ def add_lazy_oa_cuts(self, target_model, dual_values, solve_data, config, opt,
                 if constr.has_ub() \
                     and (linearize_active and abs(constr.uslack()) < config.zero_tolerance) \
                         or (linearize_violated and constr.uslack() < 0) \
-                        or (linearize_inactive and constr.uslack() > 0):
+                        or (config.linearize_inactive and constr.uslack() > 0):
```

Comment:
a bit confused by the inconsistency of the flags here.

Reply:
I think we could move all the flags to config, but to be honest, you can decide to ignore to linearize the inactive constraints, but the active and violated constraint must be linearized.

Reply:
So are you advocating for removal of the other flags? Was honestly a bit curious when @RomeoV and you put those in--thought it would be for an algorithmic variation.

Reply:
My comment applies for the convex case only and if we had exact solvers. In the nonconvex case, you would get into some weird situations... e.g. if you had a nonlinear equality constraint, both inequalities will be active but you might only want to add one of them. With non-exact solvers calling a constraint active need that zero tolerance. Let's leave the flags there but maybe use as a threshold `config.zero_tolerance` instead of 0 to separate violated and inactive.